pkg/k8s/wait.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package k8s
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func WaitUntilReady(target metav1.Object, watcher watch.Interface) error {
+	for {
+		select {
+		case ev := <-watcher.ResultChan():
+			if ev.Type == watch.Error {
+				return fmt.Errorf("error waiting for ready")
+			}
+			if obj, ok := ev.Object.(metav1.Object); !ok || obj.GetUID() != target.GetUID() {
+				// event is not for the target resource
+				continue
+			}
+			switch ev.Type {
+			case watch.Added, watch.Modified:
+				// use reflection since there is no common interface
+				if readyFunc := reflect.ValueOf(ev.Object).Elem().FieldByName("Status").Addr().MethodByName("IsReady"); readyFunc.Kind() == reflect.Func {
```

Comment:
I can't find the concrete types which this is reflecting upon, so I have no way of knowing whether the reflection is valid. Any chance of a comment to make this navigable?

Reply:
actually done now